content/en/docs/concepts/workloads/controllers/daemonset.md

```diff
@@ -201,6 +201,63 @@ them according to its `updateStrategy`.
 
 You can [perform a rolling update](/docs/tasks/manage-daemon/update-daemon-set/) on a DaemonSet.
 
+While working with the rolling updates of the DaemonSets, you will come across two important parameters: `maxSurge` and `maxUnavailable`.
+
+### maxSurge
+
+If you set this parameter to a number above 0, you allow the DaemonSet controller
+to create temporary duplicate Pods during a rollout.
+Because a DaemonSet aims to have one healthy Pod per node, this field also implies
+the number of _nodes_ that can have a duplicate Pod on them at any one moment
+during the rollout.
+While performing a rolling update, Kubernetes will keep creating new pods to replace the old ones. 
+`maxSurge` helps you to regulate the surge of additional pods beyond the desired count. 
```

Comment:
please mention that maxSurge cannot be always used, if the pod requires a specific resource (eg hostPort) https://kubernetes.io/blog/2022/09/15/app-rollout-features-reach-stable/#how-use-daemonset-maxsurge. The application must allow 2 pods per node.

Since maxUnavailable is probably safer. I would mention it first and then follwoup with describing maxSurge.

Reply:
also this part from the blog is useful: 

> Kubernetes system-level components like CNI, CSI are typically run as DaemonSets. These components can have impact on the availability of the workloads if those DaemonSets go down momentarily during the upgrades. The feature allows DaemonSet pods to temporarily increase their number, thereby ensuring zero-downtime for the DaemonSets.

Reply:
If a Pod _is_ compatible with `maxSurge`, which approach would we typically recommend?

Reply:
this is the same choice people have to make when working with deployments. It is a tradeoff between resource cost vs availability and speed. If you want a quick rollout you set appropriate `maxSurge`. And if you want to bring down the resource usage and don't mind loosing availability on each node during a rollout, you set `maxUnavalable`.

Reply:
It's a similar choice, but there's a much bigger chance of a conflict because the daemonset tries to provide a node-level service, and is implemented with the expectation that there's no duplicate process / container / pod. We need to account for that difference when we explain things.

Reply:
yes, in general `maxSurge` is not recommendable, because the node-level service has to be aware of it